Makefile

```diff
@@ -10,34 +12,41 @@ dep = $(obj:.o=.d)  # one dependency file for each source
 #macos = y
 #debug = y
 #seq = y
-
-CFLAGS = -DMKL_ILP64 -m64 -I${MKLROOT}/include -I${NUMAROOT}/include -I${TBBROOT}/include -mavx2 -mfma -mf16c -fopenmp -mavx512f -Wall #-march=skylake
+SP ?=1
+UBN ?=0
+TI  ?=0
+FG  ?=1
+NB  ?=0
+CP  ?=1
+LT  ?=1
+DEFINES = -DSP=${SP} -DPOST_VALIDATION=0 -DUSE_BROADCAST_NODE=$(UBN) -DENABLE_TBB=1 -DTHREAD_INFO=$(TI)
+DEFINES+= -DTIME_FG_LOOP=1 -DCOUNT_NODES=1 -DUSE_FG=$(FG) -DNUMA_BIND=$(NB) -DCORE_PINNING=$(CP) -DUSE_LIGHTWEIGHT=$(LT)
+CFLAGS =  -DMKL_ILP64 -m64 -I${MKLROOT}/include -I${NUMAROOT}/include -I${TBBROOT}/include -mavx2 -mfma -mf16c -fopenmp -Wall -mavx512f -march=skylake
 
 ifdef debug
   OPT = -g  -fsanitize=address -O0 -fno-omit-frame-pointer
 else
   OPT = -O3 -DNDEBUG
 endif
 
-CXX ?= g++
+#CXX ?= g++
+CXX = icpc
 CC = ${CXX} -std=c++11 
-#CC = /usr/local/opt/gcc/bin/g++-7 -std=c++11
-
-LDFLAGS = -L${MKLROOT}/lib/intel64 -lmkl_intel_lp64 -lmkl_core -lmkl_gnu_thread -lpthread -lm -ldl ${NUMAROOT}/lib/libnuma.a ${TBBROOT}/lib/libtbb.a
+LDFLAGS = -L${MKLROOT}/lib/intel64 -lmkl_intel_lp64 -lmkl_core -lmkl_sequential -L${NUMAROOT}/lib -lnuma -L${TBBROOT}/lib -ltbb -L/usr/lib -liomp5
```

Comment:
Can we please not use Intel OpenMP to be on the same page? Is using mkl_sequential needed? I basically want to know what change is needed for better performance and which is not.